### PR TITLE
LUCENE-9883: Turn on ecj missingEnumCaseDespiteDefault setting

### PR DESCRIPTION
**Pull request by zacharymorn**
# Description / Solution
Turn on ecj missingEnumCaseDespiteDefault setting

Please provide a short description of the approach taken to implement your solution.

# Tests

passed existing tests

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/lucene/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Lucene maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.

### CODE REVIEW

**Comment by mocobeta** [comment]
+1, thanks @zacharymorn!
(I think we can replace all switch statements by switch expression that makes sure that all enum values are covered without human effort, when minimum supported JDK version is bumped.)
Could you add a CHANGES entry?

**Comment by zacharymorn** [comment]
Thanks for the review @mocobeta ! I just pushed a commit to add CHANGES entry as well.

> (I think we can replace all switch statements by switch expression that makes sure that all enum values are covered without human effort, when minimum supported JDK version is bumped.)

+1 to this, and this PR makes it one step closer!

File: lucene/CHANGES.txt [hunk 1]
```diff
@@ -1062,6 +1062,8 @@ Build
 
 * LUCENE-9041: Upgrade ecj to 3.19.0 to fix sporadic precommit javadoc issues (Kevin Risden)
 
+* LUCENE-9883: Turn on ecj missingEnumCaseDespiteDefault setting. (Zach Chen)
```
Comment (mocobeta) [inline lucene/CHANGES.txt:1065]: Can you move this to "Other" section for Lucene 9.0.0 (that is the next major release)?

Comment (zacharymorn) [reply lucene/CHANGES.txt]: Ops sorry. Forgot to check version earlier. I've updated it.

**Review by mocobeta — APPROVED** [review]
I have run the whole test.
```
lucene $ ./gradlew test
BUILD SUCCESSFUL in 4m 1s
```

**Review by mikemccand — COMMENTED** [review]
Looks great!  Thank you @zacharymorn!  I just left a comment wondering why we are not backporting this to 8.x too.

**Comment by mocobeta** [comment]
> I just left a comment wondering why we are not backporting this to 8.x too.

@mikemccand I'll take care this.

**Comment by mocobeta** [comment]
I've just tried to backport this to branch_8x, but found it was much harder than I thought.
Cherry-picking is not very helpful here, also quite a lot of Solr parts needs to be modified to adopt the linter.

As there is already a large gap between main and branch_8x, I wouldn't recommend to backport large patches to 8x if it is really needed, honestly...

**Comment by zacharymorn** [comment]
Thanks Michael for the comment! Yes I have considered backporting, but didn't do it because of the potential large amount of missing enums that need to be added for solr as well like Tomoko already pointed out, but also since I feel this PR is more for lint-ing enforcement for future development (against `main` branch), and it doesn't change lucene performance and features in anyway. So I feel the benefit of backporting might be limited?

**Comment by mocobeta** [comment]
OK, it doesn't seem to be a fun job anyway;  let's leave it as it is and please turn your efforts towards things you feel more enjoyable or beneficial, @zacharymorn  :)

**Comment by mikemccand** [comment]
Thank you for looking into backporting!  I think it's fine to leave this as 9.x / Lucene only.

**Pull request merged**